src/client/api/pythonApi.ts

```diff
@@ -338,7 +338,11 @@ export class InterpreterService implements IInterpreterService {
         // we want all interpreters regardless of workspace folder so call this multiple times
         const folders = this.workspace.workspaceFolders;
         const all = folders
-            ? await Promise.all(folders.map((f) => this.apiProvider.getApi().then((api) => api.getInterpreters(f.uri))))
+            ? await Promise.all(
+                  [...folders, undefined].map((f) =>
```

Comment:
Basically add another item into the array to fetch `getInterpreters` with an argument of `undefined`.

Reply:
Seems good functionally, just a minor question. Is this a big slow down issue? Thought that getInterpreters was pretty slow? Though I would assume that slowness is just on Python extension populating the initial list, so calling with a folder and undefined might not be a big diff.

Reply:
> Is this a big slow down issue?

No it isn't, python extension uses a cache, and they don't search all over again, basically this just filters stuff (if using `undefined` its a superset of workspace, hence doesn't search worksapce again.

Reply:
& even if it were a slow down, its a bug in Python extension, and we shouldn't be focused on that
RIght now this is a bug, I don't see the kernels, but python extension already has them (again, means its cached and Python shouldn't search all over again, but if they do, its their bug - unlikely to happen)